core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -712,44 +712,62 @@ public boolean isEnabled() {
         }
 
         UserDetails asUserDetails() {
-            return new UserDetailsImpl();
-        }
-
-        private final class UserDetailsImpl implements UserDetails {
+            return new UserDetailsImpl(getAuthorities2(), getPassword(), getUsername(), isAccountNonExpired(), isAccountNonLocked(), isCredentialsNonExpired(), isEnabled());
+        }
+
+        private static final class UserDetailsImpl implements UserDetails {
+            private static final long serialVersionUID = 1L;
+            private final Collection<? extends GrantedAuthority> authorities;
+            private final String password;
+            private final String username;
+            private final boolean accountNonExpired;
+            private final boolean accountNonLocked;
+            private final boolean credentialsNonExpired;
+            private final boolean enabled;
+
+            UserDetailsImpl(Collection<? extends GrantedAuthority> authorities, String password, String username, boolean accountNonExpired, boolean accountNonLocked, boolean credentialsNonExpired, boolean enabled) {
```

Comment:
I was about to suggest that for legibility the fields and ctor params and getters be reordered into a more natural sequence. I think the original was just following alphabetical order suggested by an IDE. But now I see that https://github.com/spring-projects/spring-security/blob/cc6010876f7ee9be748471f4987b2112160bed28/core/src/main/java/org/springframework/security/core/userdetails/UserDetails.java#L45-L92 is using the same sequence!